0012-BIRS-distributed-scheduler.md

```diff
@@ -0,0 +1,463 @@
+service SchedulerCallback {
+
+    // Callback RPC for job schedule being at 'trigger' time
+    rpc TriggerJob(TriggerRequest) returns (google.protobuf.Empty) {}
+}
+
+// TriggerRequest is the message request to convey that the job is at its 'trigger' time. 
+message TriggerRequest {
+    // The job. 
+    Job job = 1; 
+
+    // The metadata associated with the job.
+    // This can contain the generated `key` for the optional state store when the daprd sidecar needs to lookup the entire data from a state store.
+    // The sidecar will use the metadata in order to know whether this job is registered for an actor. This is needed, as the routing mechanism for actors is different for the callback.
+    map<string,string> metadata = 2;
+}
+```
+
+![CallBack](./resources/0004-BIRS-distributed-scheduler/callback.png)
+
+### Scheduler Service
+
+![Pluggable Scheduler Service](./resources/0004-BIRS-distributed-scheduler/pluggableSchedulerService.png)
+
+A new `Scheduler Service` is created in the control plane. This Scheduler Service will include an embedded etcd instance (persisted) as well as a Scheduler Dapr API, which will live in `dapr/proto/scheduler/scheduler.proto`. The Scheduler Service is pluggable and allows for different implementations as needed. It is installed by default into the local development environment on `dapr init` similar to other control plane services in Dapr. This is an optional service and runs in a local container.
+
+To guarantee we don't have several Scheduler Service instances firing off the same job, we will have **virtual partitioning** (in-memory) such that each Scheduler Service instance owns a subset of all the jobs that exist. 
+
+### CLI
+
+`dapr schedule <name> –schedule “@hourly” –data “<cloud-event-json>”`
+
+### Implications
+
+- The Scheduler Building Block and Service will result in the ***deprecation*** of `actor reminders` and the `bindings.cron` component.
+
+## Expectations and alternatives
+
+* What is in scope for this proposal?
+    * To start, this will implement generic scheduler logic, then will be expanded to enable the:
+        * Delayed PubSub
+        * Scheduled Service Invocation
+        * Actor Reminders
+* What alternatives have been considered, and why do they not solve the problem?
+    * Placement with Actor reminders is *very* Actor specific. The goal is to have a new scheduler in Dapr that is reusable by several building blocks.
+* What advantages does this proposal have? 
+    * This design has the advantage of enabling the Scheduler Service to be implemented in different ways. 
+    * This design also enables flexibility in where the data is stored. Whether that is in etcd in full, or partially with a reference to a state store (component) of the user's choice.
+
+### Acceptance Criteria
+
+* How will success be measured? 
+    * POCs will be done to guarantee the optimal performance solution for the Scheduler Service, testing with 3 & 5 Scheduler Service instances
```

Comment:
Normally, 5 for an etcd cluster is considered a bit of an overkill as it offers more resiliency (can tolerate the failure of 2 nodes instead of just 1) but not necessarily better performance. In fact, compared to 3 instances, [it may even be slower when writing](https://stackoverflow.com/questions/69178090/does-scaling-etcd-affect-write-performance)

Reply:
More resiliency may be worth the perf impact for some users, so I think this should still be considered.